python/triqs_dft_tools/sumk_dft_tools.py

```diff
@@ -831,7 +831,7 @@ def spaghettis(self, broadening=None, plot_shift=0.0, plot_range=None, ishell=No
             subgrp=self.bands_data, things_to_read=things_to_read)
         if not value_read:
             return value_read
-        if ishell:
+        if ishell is not None:
```

Comment:
Should we maybe also check if `ishell` is actually a valid integer in the range allowed?